### PR TITLE
Fix build warnings of logging library

### PR DESCRIPTION
**Pull request by aggarw13**
*Issue #, if available:*

*Description of changes:*
Fix build errors for `iot_logging.c` file
Create separate file for storing logging level macros to allow inclusion in `iot_logginc.c`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/utilities/include/iot_logging_levels.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+/**
+ * @file iot_logging_levels.h
+ * @brief Defines the logging level macros.
+ */
+
+#ifndef IOT_LOGGING_LEVELS_H_
+#define IOT_LOGGING_LEVELS_H_
+
+/* The config header is always included first. */
+#include "config.h"
```
Comment (sarenameas) [inline libraries/standard/utilities/include/iot_logging_levels.h:32]: Do we need the config header here?

File: platform/posix/iot_logging.c [hunk 2]
```diff
@@ -28,10 +28,15 @@
...
 
 /* Platform clock include. */
-#include "platform/iot_clock.h"
+#include "platform/include/iot_clock.h"
+
+/* Include header for logging level macros. */
+#include "iot_logging_levels.h"
 
 /* Logging includes. */
 #include "iot_logging.h"
```
Comment (sarenameas) [inline platform/posix/iot_logging.c:42]: platform/include/iot_logging.h?
I mean they are in the same place?

File: platform/posix/iot_logging.c [hunk 3]
```diff
@@ -28,10 +28,15 @@
 /* Standard includes. */
 #include <stdarg.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
+#include <assert.h>
 
 /* Platform clock include. */
-#include "platform/iot_clock.h"
+#include "platform/include/iot_clock.h"
```
Comment (sarenameas) [inline platform/posix/iot_logging.c:36]: Does iot_clock.h need the "config.h" header?

File: libraries/standard/utilities/include/iot_logging_levels.h (hunk 1 above)
Comment (aggarw13) [reply libraries/standard/utilities/include/iot_logging_levels.h:32]: Good catch. Removed

File: platform/posix/iot_logging.c (hunk 2 above)
Comment (aggarw13) [reply platform/posix/iot_logging.c:42]: Updated

File: platform/posix/iot_logging.c (hunk 3 above)
Comment (aggarw13) [reply platform/posix/iot_logging.c:36]: Removed.

**Pull request merged**